lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -38,15 +38,24 @@ def as_json
 
         private
 
-        def data_for(serializer)
-          if serializer.respond_to?(:each)
+        def data_for(association)
+          if belongs_to_with_foreign_key?(association)
+            id = @object.send(association.reflection.foreign_key)
+            type = association.reflection.type.to_s
```

Comment:
i think this block should be extracted out to a method. So, if we wanted, we could test the construction of the ResourceIdentifier.
Though, This may be an opportunity for polymorphic build_resource_identifier later down the road